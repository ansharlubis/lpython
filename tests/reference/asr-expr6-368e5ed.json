{
    "basename": "asr-expr6-368e5ed",
    "cmd": "lpython --show-asr --no-color {infile} -o {outfile}",
    "infile": "tests/expr6.py",
    "infile_hash": "1f3b5a7d997851264e679d58353346835eb450c608f6da7d2f5e5cd2",
    "outfile": null,
    "outfile_hash": null,
    "stdout": "asr-expr6-368e5ed.stdout",
    "stdout_hash": "1575fe88f5b6f9aa891f5d4b46649ccac88b4b68d1089d0d625906b8",
    "stderr": null,
    "stderr_hash": null,
    "returncode": 0
}