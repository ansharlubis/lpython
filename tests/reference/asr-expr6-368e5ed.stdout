([1m[35mTranslationUnit[39m[0m 
   ([33mSymbolTable[39m 
      1 
      {
         test_ifexp: 
            ([1m[35mSubroutine[39m[0m 
               ([33mSymbolTable[39m 
                  2 
                  {
                     a: 
                        ([1m[35mVariable[39m[0m 
                           2 
                           a [1m[32m
                           Local[39m[0m () () [1m[32m
                           Default[39m[0m 
                           ([1m[35mInteger[39m[0m 4 []) [1m[32m
                           Source[39m[0m [1m[32m
                           Public[39m[0m [1m[32m
                           Required[39m[0m .false.), 
                     b: 
                        ([1m[35mVariable[39m[0m 
                           2 
                           b [1m[32m
                           Local[39m[0m () () [1m[32m
                           Default[39m[0m 
                           ([1m[35mInteger[39m[0m 4 []) [1m[32m
                           Source[39m[0m [1m[32m
                           Public[39m[0m [1m[32m
                           Required[39m[0m .false.), 
                     c: 
                        ([1m[35mVariable[39m[0m 
                           2 
                           c [1m[32m
                           Local[39m[0m () () [1m[32m
                           Default[39m[0m 
                           ([1m[35mLogical[39m[0m 4 []) [1m[32m
                           Source[39m[0m [1m[32m
                           Public[39m[0m [1m[32m
                           Required[39m[0m .false.)
                  }) 
               test_ifexp [] [
               ([1m[35m=[39m[0m 
                  ([1m[35mVar[39m[0m 2 [33ma[39m) ([1m[35mIntegerConstant[39m[0m [36m2[39m 
                  ([1m[35mInteger[39m[0m 4 [])) ()) 
               ([1m[35m=[39m[0m 
                  ([1m[35mVar[39m[0m 2 [33mb[39m) 
                  ([1m[35mIfExp[39m[0m 
                     ([1m[35mIntegerCompare[39m[0m 
                        ([1m[35mVar[39m[0m 2 [33ma[39m) [1m[32m
                        Eq[39m[0m ([1m[35mIntegerConstant[39m[0m [36m2[39m 
                        ([1m[35mInteger[39m[0m 4 [])) 
                        ([1m[35mLogical[39m[0m 4 []) ()) ([1m[35mIntegerConstant[39m[0m [36m6[39m 
                     ([1m[35mInteger[39m[0m 4 [])) ([1m[35mIntegerConstant[39m[0m [36m8[39m 
                     ([1m[35mInteger[39m[0m 4 [])) 
                     ([1m[35mInteger[39m[0m 4 []) ()) ()) 
               ([1m[35m=[39m[0m 
                  ([1m[35mVar[39m[0m 2 [33mc[39m) 
                  ([1m[35mIfExp[39m[0m 
                     ([1m[35mIntegerCompare[39m[0m 
                        ([1m[35mVar[39m[0m 2 [33mb[39m) [1m[32m
                        Gt[39m[0m ([1m[35mIntegerConstant[39m[0m [36m5[39m 
                        ([1m[35mInteger[39m[0m 4 [])) 
                        ([1m[35mLogical[39m[0m 4 []) ()) 
                     ([1m[35mLogicalConstant[39m[0m .true. 
                        ([1m[35mLogical[39m[0m 4 [])) 
                     ([1m[35mLogicalConstant[39m[0m .false. 
                        ([1m[35mLogical[39m[0m 4 [])) 
                     ([1m[35mLogical[39m[0m 4 []) ()) ())] [1m[32m
               Source[39m[0m [1m[32m
               Public[39m[0m [1m[32m
               Implementation[39m[0m () .false. .false.)
      }) 
   [])
(TranslationUnit (SymbolTable 1 {main_program: (Program (SymbolTable 3 {}) main_program [] []), test_ifexp: (Subroutine (SymbolTable 2 {a: (Variable 2 a Local () () Default (Integer 4 []) Source Public Required .false.), b: (Variable 2 b Local () () Default (Integer 4 []) Source Public Required .false.), c: (Variable 2 c Local () () Default (Logical 4 []) Source Public Required .false.)}) test_ifexp [] [(= (Var 2 a) (IntegerConstant 2 (Integer 4 [])) ()) (= (Var 2 b) (IfExp (IntegerCompare (Var 2 a) Eq (IntegerConstant 2 (Integer 4 [])) (Logical 4 []) ()) (IntegerConstant 6 (Integer 4 [])) (IntegerConstant 8 (Integer 4 [])) (Integer 4 []) ()) ()) (= (Var 2 c) (IfExp (IntegerCompare (Var 2 b) Gt (IntegerConstant 5 (Integer 4 [])) (Logical 4 []) ()) (LogicalConstant .true. (Logical 4 [])) (LogicalConstant .false. (Logical 4 [])) (Logical 4 []) ()) ())] Source Public Implementation () .false. .false.)}) [])
